openresty-patches/patches/1.21.4.1/ngx_lua-0.10.21_02-dyn_upstream_keepalive.patch

```diff
@@ -1,28 +1,15 @@
-From 2d12ac3e4045258b7a174b0505d92f63c26d82fc Mon Sep 17 00:00:00 2001
-From: Thibault Charbonnier <thibaultcha@me.com>
-Date: Tue, 17 Sep 2019 11:43:44 -0700
-Subject: [PATCH 1/3] feature: implemented keepalive pooling in
- 'balancer_by_lua*'.
-
----
- src/ngx_http_lua_balancer.c | 738 ++++++++++++++++++++++++++++++------
- src/ngx_http_lua_common.h   |   4 +
- src/ngx_http_lua_module.c   |   3 +
- 3 files changed, 629 insertions(+), 116 deletions(-)
-
-diff --git a/ngx_lua-0.10.21/src/ngx_http_lua_balancer.c b/ngx_lua-0.10.21/src/ngx_http_lua_balancer.c
-index f71a3e00..0d403716 100644
---- a/ngx_lua-0.10.21/src/ngx_http_lua_balancer.c
-+++ b/ngx_lua-0.10.21/src/ngx_http_lua_balancer.c
-@@ -16,46 +16,102 @@
+diff -ruN a/ngx_lua-0.10.21/src/ngx_http_lua_balancer.c b/ngx_lua-0.10.21/src/ngx_http_lua_balancer.c
+--- a/ngx_lua-0.10.21/src/ngx_http_lua_balancer.c	2022-12-02 10:58:50.054203731 +0800
++++ b/ngx_lua-0.10.21/src/ngx_http_lua_balancer.c	2022-12-05 18:22:15.351308080 +0800
+@@ -16,46 +16,104 @@
  #include "ngx_http_lua_directive.h"
-
-
+ 
+ 
 +typedef struct {
 +    ngx_uint_t                               size;
 +    ngx_uint_t                               connections;
```

Comment:
I think no need to change this, ffi use `unsigned int` because c style, It's better to use `ngx_uint_t` in the Nginx data struct, this is Nginx style.